vms/components/core/snowman_vm.go

```diff
@@ -115,8 +115,8 @@ func (svm *SnowmanVM) SetDBInitialized() error {
 }
 
 // SaveBlock saves [block] to state
-func (svm *SnowmanVM) SaveBlock(db database.Database, block snowman.Block) error {
-	return svm.State.Put(db, state.BlockTypeID, block.ID(), block)
+func (svm *SnowmanVM) SaveBlock(block snowman.Block) error {
+	return svm.State.Put(svm.DB, state.BlockTypeID, block.ID(), block)
```

Comment:
Yeah I agree that the core snowman stuff is pretty clunky and we should fix it.
The easier thing to do is just fix this in timestampVM, so that's what I did for the time being.